### PR TITLE
Add `JaCoCo` plugin

### PR DESCRIPTION
**Pull request by Oleh-Stefanovych**
This PR adds a code coverage library [JaCoCo](https://www.eclemma.org/jacoco/).

The result of checking code coverage in the `server` module at the time of writing this description:
image

### CODE REVIEW

File: server/build.gradle [hunk 1]
```diff
@@ -1,6 +1,24 @@
 plugins {
     id 'java'
     id "net.ltgt.errorprone" version "$errorpronePluginVersion"
+    id 'jacoco'
```
Comment (okucheruk) [inline server/build.gradle]: Let's have a single point of access to all of the reports generated by a build and integrate [The Build Dashboard Plugin](https://docs.gradle.org/current/userguide/build_dashboard_plugin.html)

File: gradle.properties [hunk 2]
```diff
@@ -3,8 +3,9 @@ errorproneCoreVersion=2.3.3
 errorpronePluginVersion=0.8.1
 slf4jVersion=1.7.28
 guavaVersion=28.1-jre
-jUnitVersion = 5.5.2
+jUnitVersion=5.5.2
 truthCoreVersion=1.0
+jaCoCoVersion=0.8.4
```
Comment (okucheruk) [inline gradle.properties]: Let's improve readability and use `jacocoVersion` instead.

File: server/build.gradle [hunk 3]
```diff
@@ -1,6 +1,24 @@
...
+    toolVersion = "$jaCoCoVersion"
+}
+
+jacocoTestReport {
+    reports {
+        xml.enabled false
+        csv.enabled false
+        html.destination file("${buildDir}/jacocoHtml")
+    }
+}
```
Comment (okucheruk) [inline server/build.gradle]: Let's configure violation rules also. E.g. setup minimum test coverages percentage for the `CLASS` and `BUNDLE` elements.

File: server/build.gradle (hunk 1 above)
Comment (Oleh-Stefanovych) [reply server/build.gradle]: Let's have this feature as needed in the future.

File: server/build.gradle (hunk 3 above)
Comment (Oleh-Stefanovych) [reply server/build.gradle]: Let's have this feature as needed in the future.

**Pull request merged**